### ISSUE TITLE
Mapping metadata (Extensions) between nuklei

### ISSUE THREAD

**Issue by dpwspoon** [reaktivity/ignition#8]
We should have a concept of mapping extension metadata from one form to another.  This would be useful where SOCKS has Socks extension data that aligns with either TcpExtensionData, UdpExtensionData, etc.

```
   TCP ----TcpExtension--transformed--SocksExtension--> SOCKS
```

**Comment by jfallows** [comment]
Agreed, as we start to combine nuklei together in various ways it becomes necessary to efficiently map type-specific extension metadata for `BEGIN`, `DATA` and `END` frames.

This concept of combining nuklei of different types can draw on the analogy of nuclear fusion.

Recommend we introduce the concept of a `Fusor` as a lightweight nukleus with 1-to-1 mapping of differently typed stream lifecycles plus mapping of typed extension metadata.

See https://en.wikipedia.org/wiki/Fusor.
